lua/telescope/builtin/__git.lua

```diff
@@ -193,6 +194,112 @@ git.bcommits = function(opts)
     :find()
 end
 
+git.bcommits_range = function(opts)
+  opts.current_line = (opts.current_file == nil) and get_current_buf_line(opts.winnr) or nil
+  opts.current_file = vim.F.if_nil(opts.current_file, vim.api.nvim_buf_get_name(opts.bufnr))
+  opts.entry_maker = vim.F.if_nil(opts.entry_maker, make_entry.gen_from_git_commits(opts))
+  opts.git_command = vim.F.if_nil(
+    opts.git_command,
+    git_command({ "log", "--pretty=oneline", "--abbrev-commit", "--no-patch", "-L" }, opts)
+  )
+  local visual = string.find(vim.fn.mode(), "[vV]") ~= nil
+
+  local line_number_first = opts.first
+  local line_number_last = vim.F.if_nil(opts.last, line_number_first)
+  if visual then
+    line_number_first = vim.F.if_nil(line_number_first, vim.fn.line "v")
+    line_number_last = vim.F.if_nil(line_number_last, vim.fn.line ".")
+  elseif opts.operator then
+    opts.operator = false
+    opts.operator_callback = true
+    operators.run_operator(git.bcommits_range, opts)
+    return
+  elseif opts.operator_callback then
+    line_number_first = vim.fn.line "'["
+    line_number_last = vim.fn.line "']"
+  elseif line_number_first == nil then
+    line_number_first = vim.F.if_nil(line_number_first, vim.fn.line ".")
+    line_number_last = vim.F.if_nil(line_number_last, vim.fn.line ".")
+  end
+  local line_range =
+    string.format("%d,%d:%s", line_number_first, line_number_last, Path:new(opts.current_file):make_relative(opts.cwd))
+
+  pickers
+    .new(opts, {
+      prompt_title = "Git BCommits in range",
+      finder = finders.new_oneshot_job(
+        vim.tbl_flatten {
+          opts.git_command,
+          line_range,
+        },
+        opts
+      ),
+      previewer = {
+        previewers.git_commit_diff_to_parent.new(opts),
+        previewers.git_commit_diff_to_head.new(opts),
+        previewers.git_commit_diff_as_was.new(opts),
+        previewers.git_commit_message.new(opts),
+      },
+      sorter = conf.file_sorter(opts),
+      attach_mappings = function()
```

Comment:
I think the `attach_mappings` here is identical to that for `bcommits` right? Is it possible to extract this as it's own function closing over `opts` and share this new function between the two pickers?

Reply:
Yes, they are identical. I suppose a unified `attach_mappings` function makes sense as `bcommits` and `bcommits_range` should be identical aside from the way each generates its list of commits. The only difference between the pickers should be the `prompt_title` and `finder`.

Reply:
Given that almost all of the picker is shared, I wonder if it makes sense to extract the `pickers.new()` call into a shared function that accepts `opts`, `prompt_title`, and `finder`? Those are the only aspects where the pickers differ, and it would ensure that changes to one picker also show up in the other.

Reply:
I added a commit that creates the bcommits picker in a shared function between `bcommits` and `bcommits_range`. What do you think?

Reply:
Sounds and looks good :+1: